.github/workflows/feature-matrix-updater.yaml

```diff
@@ -0,0 +1,47 @@
+name: Run Go Program and Create PR
+on:
+  push:
+    branches:
+      - main
```

Comment:
Do we want this action to run on each push to main, or something else like manually from the GitHub UI, on a schedule, etc?